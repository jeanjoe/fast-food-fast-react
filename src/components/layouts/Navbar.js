import React, { Component } from "react";
import { Link } from "react-router-dom";
import { Container, Menu, Button, Icon } from "semantic-ui-react";
import logo from "../../logo.svg";

export class Navbar extends Component {
  state = {};

  handleItemClick = (e, { name }) => this.setState({ activeItem: name });

  render() {
    const color = "red";
    return (
      <Menu stackable inverted color={color}>
        <Container>
          <Menu.Item header as={Link} to="/">
            <img src={logo} alt="logo" />
            FAST FOOD FAST APP
          </Menu.Item>

          <Menu.Menu position="right">
            <Menu.Item>
              <Button primary>
                {" "}
                <Icon name="signup" />
                Sign up
              </Button>
            </Menu.Item>

            <Menu.Item>
              <Button>
                <Icon name="sign in alternate" /> Log-in
              </Button>
            </Menu.Item>
          </Menu.Menu>
        </Container>
      </Menu>
    );
  }
}

export default Navbar;
