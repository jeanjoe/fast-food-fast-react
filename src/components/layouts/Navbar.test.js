import React from "react";
import expect from "expect";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import Navbar from "./Navbar";
configure({ adapter: new Adapter() });

describe("Test article component", () => {
  const props = {
    message: "error",
    type: "positive"
  };
  const wrapper = shallow(<Navbar {...props} />);

  it("Matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("handle handleLogout", () => {
    wrapper.instance().handleLogout();
  });
});
