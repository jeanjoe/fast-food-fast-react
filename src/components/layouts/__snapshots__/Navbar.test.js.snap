// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test article component Matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    message="error"
    type="positive"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Container>
        <MenuItem
          as={[Function]}
          header={true}
          to="/"
        >
          <img
            alt="logo"
            src="logo.svg"
          />
          FAST FOOD FAST APP
        </MenuItem>
        <MenuMenu
          position="right"
        >
          <React.Fragment>
            <MenuItem>
              <Button
                as={[Function]}
                primary={true}
                to="/sign-up"
              >
                <Icon
                  as="i"
                  name="signup"
                />
                Sign up
              </Button>
            </MenuItem>
            <MenuItem>
              <Button
                as={[Function]}
                to="/sign-in"
              >
                <Icon
                  as="i"
                  name="sign in alternate"
                />
                 Log-in
              </Button>
            </MenuItem>
          </React.Fragment>
        </MenuMenu>
      </Container>,
      "color": "red",
      "inverted": true,
      "stackable": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <MenuItem
            as={[Function]}
            header={true}
            to="/"
          >
            <img
              alt="logo"
              src="logo.svg"
            />
            FAST FOOD FAST APP
          </MenuItem>,
          <MenuMenu
            position="right"
          >
            <React.Fragment>
              <MenuItem>
                <Button
                  as={[Function]}
                  primary={true}
                  to="/sign-up"
                >
                  <Icon
                    as="i"
                    name="signup"
                  />
                  Sign up
                </Button>
              </MenuItem>
              <MenuItem>
                <Button
                  as={[Function]}
                  to="/sign-in"
                >
                  <Icon
                    as="i"
                    name="sign in alternate"
                  />
                   Log-in
                </Button>
              </MenuItem>
            </React.Fragment>
          </MenuMenu>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": [Function],
            "children": Array [
              <img
                alt="logo"
                src="logo.svg"
              />,
              "FAST FOOD FAST APP",
            ],
            "header": true,
            "to": "/",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "FAST FOOD FAST APP",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <React.Fragment>
              <MenuItem>
                <Button
                  as={[Function]}
                  primary={true}
                  to="/sign-up"
                >
                  <Icon
                    as="i"
                    name="signup"
                  />
                  Sign up
                </Button>
              </MenuItem>
              <MenuItem>
                <Button
                  as={[Function]}
                  to="/sign-in"
                >
                  <Icon
                    as="i"
                    name="sign in alternate"
                  />
                   Log-in
                </Button>
              </MenuItem>
            </React.Fragment>,
            "position": "right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <MenuItem>
                  <Button
                    as={[Function]}
                    primary={true}
                    to="/sign-up"
                  >
                    <Icon
                      as="i"
                      name="signup"
                    />
                    Sign up
                  </Button>
                </MenuItem>,
                <MenuItem>
                  <Button
                    as={[Function]}
                    to="/sign-in"
                  >
                    <Icon
                      as="i"
                      name="sign in alternate"
                    />
                     Log-in
                  </Button>
                </MenuItem>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Button
                    as={[Function]}
                    primary={true}
                    to="/sign-up"
                  >
                    <Icon
                      as="i"
                      name="signup"
                    />
                    Sign up
                  </Button>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": [Function],
                    "children": Array [
                      <Icon
                        as="i"
                        name="signup"
                      />,
                      "Sign up",
                    ],
                    "primary": true,
                    "to": "/sign-up",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "as": "i",
                        "name": "signup",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "Sign up",
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Button
                    as={[Function]}
                    to="/sign-in"
                  >
                    <Icon
                      as="i"
                      name="sign in alternate"
                    />
                     Log-in
                  </Button>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": [Function],
                    "children": Array [
                      <Icon
                        as="i"
                        name="sign in alternate"
                      />,
                      " Log-in",
                    ],
                    "to": "/sign-in",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "as": "i",
                        "name": "sign in alternate",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    " Log-in",
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": Symbol(react.fragment),
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Container>
          <MenuItem
            as={[Function]}
            header={true}
            to="/"
          >
            <img
              alt="logo"
              src="logo.svg"
            />
            FAST FOOD FAST APP
          </MenuItem>
          <MenuMenu
            position="right"
          >
            <React.Fragment>
              <MenuItem>
                <Button
                  as={[Function]}
                  primary={true}
                  to="/sign-up"
                >
                  <Icon
                    as="i"
                    name="signup"
                  />
                  Sign up
                </Button>
              </MenuItem>
              <MenuItem>
                <Button
                  as={[Function]}
                  to="/sign-in"
                >
                  <Icon
                    as="i"
                    name="sign in alternate"
                  />
                   Log-in
                </Button>
              </MenuItem>
            </React.Fragment>
          </MenuMenu>
        </Container>,
        "color": "red",
        "inverted": true,
        "stackable": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <MenuItem
              as={[Function]}
              header={true}
              to="/"
            >
              <img
                alt="logo"
                src="logo.svg"
              />
              FAST FOOD FAST APP
            </MenuItem>,
            <MenuMenu
              position="right"
            >
              <React.Fragment>
                <MenuItem>
                  <Button
                    as={[Function]}
                    primary={true}
                    to="/sign-up"
                  >
                    <Icon
                      as="i"
                      name="signup"
                    />
                    Sign up
                  </Button>
                </MenuItem>
                <MenuItem>
                  <Button
                    as={[Function]}
                    to="/sign-in"
                  >
                    <Icon
                      as="i"
                      name="sign in alternate"
                    />
                     Log-in
                  </Button>
                </MenuItem>
              </React.Fragment>
            </MenuMenu>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": [Function],
              "children": Array [
                <img
                  alt="logo"
                  src="logo.svg"
                />,
                "FAST FOOD FAST APP",
              ],
              "header": true,
              "to": "/",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "logo",
                  "src": "logo.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "FAST FOOD FAST APP",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <React.Fragment>
                <MenuItem>
                  <Button
                    as={[Function]}
                    primary={true}
                    to="/sign-up"
                  >
                    <Icon
                      as="i"
                      name="signup"
                    />
                    Sign up
                  </Button>
                </MenuItem>
                <MenuItem>
                  <Button
                    as={[Function]}
                    to="/sign-in"
                  >
                    <Icon
                      as="i"
                      name="sign in alternate"
                    />
                     Log-in
                  </Button>
                </MenuItem>
              </React.Fragment>,
              "position": "right",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <MenuItem>
                    <Button
                      as={[Function]}
                      primary={true}
                      to="/sign-up"
                    >
                      <Icon
                        as="i"
                        name="signup"
                      />
                      Sign up
                    </Button>
                  </MenuItem>,
                  <MenuItem>
                    <Button
                      as={[Function]}
                      to="/sign-in"
                    >
                      <Icon
                        as="i"
                        name="sign in alternate"
                      />
                       Log-in
                    </Button>
                  </MenuItem>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Button
                      as={[Function]}
                      primary={true}
                      to="/sign-up"
                    >
                      <Icon
                        as="i"
                        name="signup"
                      />
                      Sign up
                    </Button>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "as": [Function],
                      "children": Array [
                        <Icon
                          as="i"
                          name="signup"
                        />,
                        "Sign up",
                      ],
                      "primary": true,
                      "to": "/sign-up",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "as": "i",
                          "name": "signup",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "Sign up",
                    ],
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Button
                      as={[Function]}
                      to="/sign-in"
                    >
                      <Icon
                        as="i"
                        name="sign in alternate"
                      />
                       Log-in
                    </Button>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "as": [Function],
                      "children": Array [
                        <Icon
                          as="i"
                          name="sign in alternate"
                        />,
                        " Log-in",
                      ],
                      "to": "/sign-in",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "as": "i",
                          "name": "sign in alternate",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      " Log-in",
                    ],
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": Symbol(react.fragment),
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
